### PR TITLE
feat: inject kubernetes discovery environment variable

### LINKED ISSUE #6830: bug: k8s service discovery error: not found environment variable KUBERNETES_SERVICE_HOST
### Current Behavior

k8s discovery can't get KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT from ENV

but in pod I can echo KUBERNETES_SERVICE_HOST var.

image

### Expected Behavior

_No response_

### Error Logs

2022/04/12 07:22:00 [error] 48#48: init_worker_by_lua error: /usr/local/apisix/apisix/discovery/kubernetes/init.lua:342: not found environment variable KUBERNETES_SERVICE_HOST
stack traceback:
        [C]: in function 'error'
        /usr/local/apisix/apisix/discovery/kubernetes/init.lua:342: in function 'init_worker'
        /usr/local/apisix/apisix/discovery/init.lua:35: in function 'init_worker'
        /usr/local/apisix/apisix/init.lua:109: in function 'http_init_worker'
        init_worker_by_lua:2: in main chunk

### Steps to Reproduce

1.install apisix use helm [apisix-helm-chart](https://github.com/apache/apisix-helm-chart/blob/master/docs/en/latest/apisix.md)
2.add discovery:kubernetes: { } in apisix configmap
3.k logs -f po

### Environment

- APISIX version (run `apisix version`):2.13.0
- Operating system (run `uname -a`):Linux hecs-x-large-4-linux-20200603165605 3.10.0-1160.53.1.el7.x86_64 #1 SMP Fri Jan 14 13:59:45 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

### PR DESCRIPTION
**Pull request by zhixiongdu027**
### Description

Fix #6830 

Inject the env variable when k8s service discovery is enabled -> apisix/opt.lua

### Checklist

- [Y] I have explained the need for this PR and the problem it solves
- [Y] I have explained the changes or the new features added to this PR
- [Y] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by zhixiongdu027** [comment]
@spacewander @tzssangglass 
pr had ready for review 
Please take a look ，tks

File: t/cli/test_kubernetes.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+. ./t/cli/common.sh
+
+exit_if_not_customed_nginx
```
Comment (spacewander) [inline t/cli/test_kubernetes.sh:22]: We can run this test with regular Nginx?

Comment (zhixiongdu027) [reply t/cli/test_kubernetes.sh:22]: Sorry, I don't quite get what you mean.

But I think the running environment of this test file should be the same as other test files in the same directory.

Comment (zhixiongdu027) [reply t/cli/test_kubernetes.sh:22]: > We can run this test with regular Nginx?

Sorry, I don't quite get what you mean.

But I think the running environment of this test file should be the same as other test files in the same directory.

Comment (zhixiongdu027) [reply t/cli/test_kubernetes.sh:22]: > > We can run this test with regular Nginx?
> 
> Sorry, I don't quite get what you mean.
> 
> But I think the running environment of this test file should be the same as other test files in the same directory.

I got ， and will fix

**Review by spacewander — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**